src/Simplex/Messaging/Agent/Store/SQLite.hs

```diff
@@ -649,25 +676,145 @@ instance (MonadUnliftIO m, MonadError StoreError m) => MonadAgentStore SQLiteSto
         (provider, token, host, port)
 
   getNtfSubscription :: SQLiteStore -> RcvQueue -> m (Maybe NtfSubscription)
-  getNtfSubscription _st _rcvQueue = throwError SENotImplemented
+  getNtfSubscription st RcvQueue {server = smpServer@ProtocolServer {host, port}, rcvId} =
+    liftIO . withTransaction st $ \db ->
+      ntfSubscription
+        <$> DB.query
+          db
+          [sql|
+            SELECT s.ntf_host, s.ntf_port, s.ntf_key_hash,
+              ns.smp_ntf_id, ns.ntf_sub_id, ns.ntf_sub_status, ns.ntf_sub_action_ts
+            FROM ntf_subscriptions ns
+            JOIN ntf_servers s USING (ntf_host, ntf_port)
+            WHERE ns.smp_host = ? AND ns.smp_port = ? AND ns.smp_rcv_id = ?
+          |]
+          (host, port, rcvId)
+    where
+      ntfSubscription [(ntfHost, ntfPort, ntfKeyHash, ntfQueueId, ntfSubId, ntfSubStatus, ntfSubActionTs)] =
+        let ntfServer = ProtocolServer ntfHost ntfPort ntfKeyHash
+         in Just NtfSubscription {smpServer, rcvQueueId = rcvId, ntfQueueId, ntfServer, ntfSubId, ntfSubStatus, ntfSubActionTs}
+      ntfSubscription _ = Nothing
 
-  createNtfSubscription :: SQLiteStore -> NtfSubscription -> m ()
-  createNtfSubscription _st _ntfSub = throwError SENotImplemented
+  createNtfSubscription :: SQLiteStore -> NtfSubscription -> NtfSubOrSMPAction -> m ()
+  createNtfSubscription st NtfSubscription {smpServer = (SMPServer host port _), rcvQueueId, ntfQueueId, ntfServer = (SMPServer ntfHost ntfPort _), ntfSubId, ntfSubStatus, ntfSubActionTs} ntfAction =
+    liftIO . withTransaction st $ \db ->
+      case ntfAction of
+        NtfSubSMPAction nsa ->
+          DB.execute
+            db
+            [sql|
+              INSERT INTO ntf_subscriptions
+                (smp_host, smp_port, smp_rcv_id, smp_ntf_id, ntf_host, ntf_port, ntf_sub_id, ntf_sub_status, ntf_sub_smp_action, ntf_sub_action_ts) VALUES (?,?,?,?,?,?,?,?,?,?)
+            |]
+            (host, port, rcvQueueId, ntfQueueId, ntfHost, ntfPort, ntfSubId, ntfSubStatus, nsa, ntfSubActionTs)
+        NtfSubAction nsa ->
+          DB.execute
+            db
+            [sql|
+              INSERT INTO ntf_subscriptions
+                (smp_host, smp_port, smp_rcv_id, smp_ntf_id, ntf_host, ntf_port, ntf_sub_id, ntf_sub_status, ntf_sub_action, ntf_sub_action_ts) VALUES (?,?,?,?,?,?,?,?,?,?)
+            |]
+            (host, port, rcvQueueId, ntfQueueId, ntfHost, ntfPort, ntfSubId, ntfSubStatus, nsa, ntfSubActionTs)
```

Comment:
this can easily be made into one query, you just need to map ntfAction value into a (Maybe, Maybe) tuple - no need to repeat the whole query